### PR TITLE
feat(executor): Support accessing output parameters by PNS executor running as non-root

### PR DESCRIPTION
**Pull request by vladlosev**
The Argo PNS executor requires the SYS_CHROOT and the SYS_PTRACE capabilities for access to the workflow output parameters when they are saved inside the image. When it is run as a root, those capabilities are available to it immediately because they are added by the workflow controller to the container's security context. But when it's run as a non-root user, the capabilities are available in the process's inherited capability set but neither in its effective nor permitted sets. Thus, a chroot attempt in such mode currently fails. This PR adds the requisite capability bits to the `argoexec` binary in the container, which allows the executor process to gain these capabilities when launched, and use those syscalls successfully.

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

**Review by alexec — COMMENTED** [review]
this is a powerful PR

File: test/e2e/run_as_not_root_test.go [hunk 1]
```diff
@@ -23,6 +23,15 @@ func (s *RunAsNonRootSuite) TestRunAsNonRootWorkflow() {
 		WaitForWorkflow(fixtures.ToBeSucceeded)
 }
 
+func (s *RunAsNonRootSuite) TestRunAsNonRootWithOutputParams() {
+	s.Need(fixtures.None(fixtures.Docker, fixtures.K8SAPI, fixtures.Kubelet))
```
Comment (alexec) [inline test/e2e/run_as_not_root_test.go:27]: minor - do you want to include `emissary` in your test? instead maybe just 

```
s.Need(fixtures.PNS)
```

File: test/e2e/smoke/runasnonroot-output-params.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (alexec) [inline test/e2e/smoke/runasnonroot-output-params.yaml:1]: please append `-pipeline.yaml` to name

Comment (vladlosev) [reply test/e2e/smoke/runasnonroot-output-params.yaml:1]: Done.

File: test/e2e/run_as_not_root_test.go (hunk 1 above)
Comment (vladlosev) [reply test/e2e/run_as_not_root_test.go:27]: The test passes under emissary, and it is a valid use case (I can attest to that!) so I think we want to keep it for emissary to ensure there are no regressions in the future. That is unless you have reasons to explicitly not support this use case for emissary. Please let me know if that's the  case and I will drop it for emissary.

**Review by alexec — APPROVED** [review]

**Pull request merged**